clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -387,13 +431,26 @@ void SYCL::x86_64::BackendCompiler::ConstructJob(Compilation &C,
   ArgStringList CmdArgs;
   CmdArgs.push_back(Args.MakeArgString(Twine("-ir=") + Output.getFilename()));
   CmdArgs.push_back("-device=cpu");
+  ArgStringList ForeachArgs;
   for (const auto &II : Inputs) {
+    if (II.getType() == types::TY_Tempfilelist) {
+      std::string FileName(II.getFilename());
```

Comment:
Suggest initializing before entering the condition and reusing [3]

Reply:
Okay.